pkg/executor/memtable_reader.go

```diff
@@ -93,6 +93,22 @@ func (*MemTableReaderExec) isInspectionCacheableTable(tblName string) bool {
 	}
 }
 
+// Open implements the Executor Open interface.
+func (e *MemTableReaderExec) Open(ctx context.Context) error {
+	err := e.BaseExecutor.Open(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Activate the transaction, otherwise SELECT .. FROM INFORMATION_SCHEMA.XX .. does not block GC worker.
+	// And if the query last too long (10min), it causes error "GC life time is shorter than transaction duration"
+	if !strings.HasPrefix(e.table.Name.L, "cluster_") {
+		// Skip cluster_ tables because it could be a tidb coprocessor request and txn is not prepared.
```

Comment:
The special checking is an indication that `MemTableReaderExec` should be separated into multiple structure. As the name `MemTableReaderExec` I think it's a pure memory implementation, and no need to consider KV transaction, GC and other things. In infoschema v2 project, `MemTableReaderExec` involves KV read/write. And as this comment said, it also involves coprocessor request. In future we'd better separate InfoSchemaV2Executor and ClusterTablesExecutor.

Reply:
This still fails to pass CI, there are more cases.
It's *toooooo difficult* to figure out all the callers of this function that `e.Ctx().Txn(true)` fails.

Reply:
I change to another way to check, hope this time no panics.